proto/encryption.proto

```diff
@@ -58,3 +58,28 @@ message GetEncryptionConfigResponse {
 
   repeated KMS supported_kms = 3;
 }
+
+// Auth info (client, group) is passed via headers.
+message GetEncryptionKeyRequest {}
+
+message EncryptionKey {
+  // Encryption keys and data are versioned to support key rotation. This
+  // version is passed to the key derivation function and encoded in the
+  // encrypted data.
+  //
+  // More about customer-managed encryption can be found in
+  // http://go/customer-managed-encryption
+  int32 version = 1;
```

Comment:
This is intended to replace this version: https://github.com/buildbuddy-io/buildbuddy/blob/caf83be60d8c0c6d883a3dc5e3586c4a5bf66384/enterprise/server/crypter_service/crypter_service.go#L48 which is passed into the key-derivation function and in an encrypted "auth header" which is verified on decryption. I think this is to support cycling keys on our end, which we'd want to control in the apps, hence its inclusion here. Maybe it should be a flag as well?